include/service/entities/article_service.js

```diff
@@ -132,11 +132,11 @@ module.exports = function ArticleServiceModule(pb) {
 
         //build out the ordering
         var order;
-        if (util.isArray(options.order)) {
+        if (util.isObject(options.order)) {
             order = options.order;
         }
         else {
-            order = [['publish_date', pb.DAO.DESC], ['created', pb.DAO.DESC]];
+            order = {'publish_date': pb.DAO.DESC, 'created': pb.DAO.DESC};
```

Comment:
That's weird that they show the [key, direction] format in the documentation. That has to be a typo. Here's what I'm seeing if I run queries directly against MongoDB, outside of PencilBlue:

MongoDB 3.0.1:
db.getCollection('article').find({}).sort(['publish_date', -1])   - Records returned in natural order, not by publish_date
db.getCollection('article').find({}).sort([['publish_date', -1]]) - Error: BadValue bad order array [2]
db.getCollection('article').find({}).sort({'publish_date': -1}) - Records returned in order of publish_date
db.getCollection('article').find({}).sort([{'publish_date': -1}]) - Records returned in order of publish_date

MongoDB 2.4.9:
db.getCollection('article').find({}).sort(['publish_date', -1])   - Records returned in natural order, not by publish_date
db.getCollection('article').find({}).sort([['publish_date', -1]]) - Records returned in natural order, not by publish_date
db.getCollection('article').find({}).sort({'publish_date': -1}) - Records returned in order of publish_date
db.getCollection('article').find({}).sort([{'publish_date': -1}]) - Records returned in natural order, not by publish_date

I agree with you on the hash key order issue. It looks like arrays work on MongoDB 3, but not MongoDB 2.4. Any suggestions on how to proceed?